rlp/list.libevm.go

```diff
@@ -75,3 +140,53 @@ func DecodeList[T any](s *Stream) ([]*T, error) {
 	})
 	return vals, err
 }
+
+// DecodeStructFields is the inverse of [EncodeStructFields]. All destination
+// fields, be they `required` or `optional`, MUST be pointers and all `optional`
+// fields MUST be provided in case they are present in the RLP being decoded.
+//
+// Typically, the arguments to this function mirror those passed to
+// [EncodeStructFields] except for being pointers. See the example.
```

Comment:
nit does `[cChainBodyExtras]` referring to a test file work? 🤔 

```suggestion
// [EncodeStructFields] except for being pointers. See the example with [cChainBodyExtras].
```

Reply:
It's not that example though, it's the one for the method so will be right there in the documentation:

![image](https://github.com/user-attachments/assets/e97aa3d4-c59b-423d-9776-ad1d0fcf88cc)